### PR TITLE
ncsdk - Movidius Neural Compute Stick SDK

### PR DESCRIPTION
**Pull request by RitwikSaikia**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by woodruffw** [comment]
```
ncsdk:
  * New formulae should not require patches to build. Patches should be submitted and accepted upstream first.
```

File: Formula/ncsdk.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        python_name = "python#{version}"
+        python_path = lib/"#{python_name}/site-packages"
+        python_path.mkpath
+
+        cp_r Dir["api/python/*"], python_path
+      end
+    end
+  end
+
+  test do
```
Comment (woodruffw) [inline Formula/ncsdk.rb]: All new formulae are required to include a substantive test (i.e., something that isn't just `--help` or `--version`, or in this case `true`).

File: Formula/ncsdk.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    sha256 "fc715b7c6226b16a7a25fe26aabf10144d2b8dc386355a589ef90f2f0330b469"
+  end
+
+  def install
+    chmod "+x", "install-homebrew.sh"
+    system "./install-homebrew.sh", prefix
+    with_python = true
+    with_python = false if build.without? "python"
+
+    if with_python
```
Comment (woodruffw) [inline Formula/ncsdk.rb]: You can get rid of the `with_python` variable:

```ruby
if build.with? "python"
  # whatever
end 
```

File: Formula/ncsdk.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  url "https://github.com/movidius/ncsdk/archive/v1.12.00.01.tar.gz"
+  sha256 "aceca6ffa87c87c2b29f5a89ed97a96f0b0238ff16d9c45e61e3e891a63c6386"
+
+  option "with-test", "Run build-time check"
+  option "without-python", "Build without python support"
+
+  depends_on "libusb"
+  depends_on "python3" => :recommended
+
+  patch do
```
Comment (fxcoudert) [inline Formula/ncsdk.rb]: `New formulae should not require patches to build. Patches should be submitted and accepted upstream first.`

**Comment by fxcoudert** [comment]
Build fails on El Capitan:

```
clang -O2 -Wall -pthread -fPIC -MMD -MP -I. -I../include -I/usr/local/include/libusb-1.0  -c mvnc_api.c -o obj-x86_64/mvnc_api.o
usb_boot.c:62:16: error: use of undeclared identifier 'CLOCK_REALTIME'
        clock_gettime(CLOCK_REALTIME, ptr);
                      ^
mvnc_api.c:102:16: error: use of undeclared identifier 'CLOCK_MONOTONIC'
        clock_gettime(CLOCK_MONOTONIC, &ts);
                      ^
```

**Comment by ilovezfs** [comment]
Please feel free to reopen when this is building on OS X 10.11 without patching.

**Pull request closed without merging**